airgun/entities/host_new.py

```diff
@@ -42,6 +43,7 @@ def get_details(self, entity_name, widget_names=None):
         view.wait_displayed()
         self.browser.plugin.ensure_page_safe()
         # Run this read twice to navigate to the page and load it before reading
+        time.sleep(2)
```

Comment:
please remove the hard sleep and use the wait_for everywhere it needed.

Reply:
i am working on it